### ISSUE TITLE
Use jekyll

### ISSUE THREAD

**Issue by jgog** [thomaspark/bootswatch#623]
Now that we are preparing for v4, I think instead of generating theme previews manually, let's generate them using something like **[jekyll](https://jekyllrb.com)**

**Comment by thomaspark** [comment]
Hey @jgog, this is much overdue! I'm thinking a grunt task using [grunt-html-build](https://www.npmjs.com/package/grunt-html-build).

**Comment by jgog** [comment]
why `grunt-html-build` when you can do a lot (maybe all) of it with **jekyll**?

**Comment by thomaspark** [comment]
Both may get the job done, but it would be one less dependency since we already use grunt for other tasks.

**Comment by jgog** [comment]
agreed, but jekyll wouldn't be a dependency, since [Github Pages](https://help.github.com/articles/using-jekyll-as-a-static-site-generator-with-github-pages/) supports it out of the box. But either way, the bottom line is that the preview page of each theme needs to be made better and generated dynamically, instead of it being inside each theme.

**Comment by IamManchanda** [comment]
jekyll would be better ,,,, simply because bootstrap also uses jekyll

**Comment by eberkund** [comment]
I prefer Hugo over Jekyll nowadays, it is much faster to build and doesn't require a whole Ruby dev environment to run.

**Comment by jenil** [comment]
Hey @thomaspark 

Congratulations on 🚀 the v4 beta.

I was wondering if we could now streamline the documentation & build efficiently using Jekyll.
Take a look at how I do it for [Bulmaswatch](https://github.com/jenil/bulmaswatch)

**Comment by thomaspark** [comment]
Hey @jenil, thanks! Next step will be to reorganize the repo and build the html from template. I'll check out your approach with Bulmaswatch.

**Comment by m5o** [comment]
@jenil Github now also offers the usage of the `master` branch for Github Pages. Did you see the blog post about it? 

:octocat:  https://github.com/blog/2233-publish-your-project-documentation-with-github-pages

**Comment by jenil** [comment]
ahh yes, but I had already started work on the gh-pages branch. Maybe I will eventually move it to `master`

**Comment by thomaspark** [comment]
Hey @m5o, per your comments in #756, agree that @jenil has a great setup going with bulmaswatch. I'm still very much interested in having the index pages generated from template.

**Comment by m5o** [comment]
> I'm still very much interested in having the index pages generated from template.

@thomaspark Only the index pages or more adoption from bulmaswatch?

**Comment by thomaspark** [comment]
@m5o, generating the index/preview pages would be a big time-saver and the priority, but what other parts of it do you think would be beneficial?

**Comment by m5o** [comment]
> but what other parts of it do you think would be beneficial?

As mentioned in #756 like using gh-pages for docs, the `gulpfile` looks decent with sourcemaps, browser-sync, the bookmarklet, the [test page](https://github.com/jenil/bulmaswatch/blob/gh-pages/test.html) (@jenil 🤷‍♀️Is it used for visual CI?)
Bigger picture: Maybe join forces and maintain a set of tools (theming scaffold) etc

**Comment by jenil** [comment]
@thomaspark you can generate both the preview pages as well as the thumbnail pages and serve the thumbs as iframes on the homepage. For the 0.6.0 version, I am also separating each section of the preview page into [include files](https://github.com/jenil/bulmaswatch/tree/0.6.0/_includes).

@m5o there is no CI at the moment, I use the test page for development.

**Comment by m5o** [comment]
@thomaspark what are your general thoughts about this? Do you think about re-use some stuff @jenil build. He made great progress on `0.6.x` releases. Move to a jekyll based docs workflow, also like bootstrap itself was doing. I would volunteer to help, to give bootswatch a nice spring cleaning 🤓

**Comment by jenil** [comment]
I'm totally up for help, but maybe in a months time after I graduate 💯

**Comment by thomaspark** [comment]
😃Congrats in advance on graduation @jenil!

**Comment by m5o** [comment]
@thomaspark Can you contribute the [bootswatch logo](https://bootswatch.com/_assets/img/logo-nav.png) as vector SVG?

**Comment by m5o** [comment]
Sorry guys, some RL stuff blocked me working on this. I still have it on my list.

**Comment by m5o** [comment]
@thomaspark I wanted to know if it is enough to concentrate only on the latest `v4.x` bootstrap themes. Or would you like to include the older versions `v3` and `v2` in the new build process?

**Comment by thomaspark** [comment]
Hey @m5o, it would be enough just for the latest `v4`.

**Comment by eberkund** [comment]
If you haven't already started @m5o consider looking at Hugo + Netlify, I use it for a number of projects and it is really amazing for static sites. Also I think Netlify has a free plan for open source projects like this one.
